python-api/livy/client.py

```diff
@@ -0,0 +1,167 @@
+
+    __CONFIG_SECTION = 'env'
+    __LIVY_CLIENT_CONF_DIR = "LIVY_CLIENT_CONF_DIR"
+
+    def __init__(self, uri, load_defaults=True, conf_dict = dict()):
+        """
+         Loads the provided config and creates a http-based client
+         :param uri: Livy server uri generated from urlparse lib
+         :type uri: ParseResult generated from urlparse lib
+
+         >>> from urlparse import urlparse
+         >>> uri = urlparse("http://test")
+
+         :param boolean load_defaults: Default value is true. If true, loads the
+             livy client configuration from the files 'livy-client.conf' and
+             'spark-defaults.conf'. The path to the files should be defined in the
+             'LIVY_CLIENT_CONF_DIR' environment variable
+         :param dict: Default value is empty dict. The key-value pairs in the dict
+             will be loaded to the config
+         :returns An instance of thsi class
+        """
+        self.__config = SafeConfigParser()
+        self.__load_config(load_defaults, conf_dict)
+        self.__set_uri(uri)
+
+    def submit(self, job):
+        """
+        Submits a job for execution to the spark cluster.
+        :param job: A function that has JobContext as the parameter. Spark jobs can
+            be created with the help of JobContext, which exposes the Spark libraries
+        :type job: A named or anonymous function that has type JobContext as the parameter
+        :returns A future to monitor the status of the job
+        """
+        pass
+
+    def run(self, job):
+        """
+        Asks the remote context to run a job immediately.
+
+        Normally, the remote context will queue jobs and execute them based on how many worker
+        threads have been configured. This method will run the submitted job in the same thread
+        processing the RPC message, so that queueing does not apply.
+
+        It's recommended that this method only be used to run code that finishes quickly. This
+        avoids interfering with the normal operation of the context.
+        :param job: A function that has JobContext as the parameter. Spark jobs can
+            be created with the help of JobContext, which exposes the Spark libraries
+        :type job: A named or anonymous function that has type JobContext as the parameter
+        :returns A future to monitor the status of the job
+        """
+        pass
+
+    def add_file(self, file_uri):
+        """
+        Adds a file to the running remote context.
+
+        Note that the URL should be reachable by the Spark driver process. If running the driver
+        in cluster mode, it may reside on a different host, meaning "file:" URLs have to exist
+        on that node (and not on the client machine).
+
+        :param file_uri: Representation of the path to a file using the filesystem encoding
+        :type file_uri: PurePath from pathlib or pathlib2(preferred)
```

Comment:
This is needlessly rigid. It is acceptable to accept either a string that can be coerced to a PurePath or a PurePath object. Perhaps:

```
file_uri : string, pathlib.PurePath, or pathlib2.PurePath
    Complete path to a file according to the filesystem encoding
```